recipes/mingw-builds/all/conanfile.py

```diff
@@ -0,0 +1,82 @@
+import os
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class MingwConan(ConanFile):
+    name = "mingw-builds"
+    description = "MinGW is a contraction of Minimalist GNU for Windows"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/ericLemanissier/mingw-builds"
+    license = "ZPL-2.1", "MIT", "GPL-2.0-or-later"
+    topics = ("gcc", "gnu", "unix", "mingw32", "binutils")
+    settings = "os", "arch"
+    options = {"threads": ["posix", "win32"], "exception": ["seh", "sjlj"]}
+    default_options = {"threads": "posix", "exception": "seh"}
+
+    @property
+    def _settings_build(self):
+        return getattr(self, "settings_build", self.settings)
+
+    def validate(self):
```

Comment:
I think validating `self.settings.arch_target`/`self.settings.os_target`/`self.settings.compiler.*` might be useful because `package_id` will force the packages used with mingw-builds as build requirement to use the mingw toolchain.